### PR TITLE
VFB-279 create correct shopping list

### PR DESCRIPTION
**Pull request by rjesson-softwire**
## What's changed
The 'download shopping list' action on the parcels page now creates a pdf that only shows items which are in the list type of the parcel.

## Screenshots / Videos
| Before     | After      |
|------------|------------|
| image | image|

## Checklist
- [x] The ticket is up-to-date - Please document any deviations from the original approach if there is any.
- [x] I have documented the testing steps for QA
- [x] I have self-reviewed this PR
- [x] Make sure you've verified it works via `npm run dev`
- [x] Make sure you've verified it works via `npm run build` and `npm run start`
- [x] Make sure you've fixed all linting problems with `npm run lint_fix`
- [x] Make sure you've tested via `npm run test`

---
## AI generated change summary

The following is a summary of the changes in the PR generated by [What The Diff](https://whatthediff.ai/).
Delete the command below if you don't want this to be generataed.

* **A New Way to Identify Lists Implemented**
  - The code in the files `fetch.ts`, `getParcelsData.ts` and `shoppingListPdfDataProps.ts` was updated to include a new reference for `ListName` from `"@/app/lists/ListDataview"`. This means the program will now be using a specific reference to identify various lists.
  
* **Changes to Shopping List PDF Representation**
  - The `ShoppingListPdf.tsx` file has undergone several changes, aimed at using a new type of item representation, identified as `ItemForPDF`, instead of the previously used `Item`. This is an enhancement to provide more detailed or specific information in the shopping list PDF document.
  
* **Parcel Information Enhancement**
  - The `getParcelsData.ts` was updated to include a new `listType` property in the `ParcelInfo` interface and the function `prepareParcelInfo` now sets this new property. This augments the parcel information with the type of list it associated with.
  
* **Filtering of Shopping List Data**
  - The code to generate shopping list data (`getShoppingListData.ts`) has been revised. Now, it includes a new filtering function that extracts items based on the type of list they belong to.
  
* **Improved Item Interface and Shopping List PDF Data**
  - The `Item` interface has been renamed and modified to `ItemWithListType` in `shoppingListPdfDataProps.ts`. Additionally, different types and functions were updated to use `ItemForPDF` or `ItemWithListType` instead of `Item`. This shift allows for a more precise representation of items in the shopping list PDF.
  
* **New Item Filtering Function**
  - A newly added function `filterItemsbyListType` in `shoppingListPdfDataProps.ts` allows filtering an array of items based on a given list type, enabling more specific data handling based on list categories.
  
These adjustments overall improve how lists and items are represented and how shopping list data is generated and manipulated, making it more efficient and aligned with specific list types.

### CODE REVIEW

File: src/pdf/ShoppingList/shoppingListPdfDataProps.ts [hunk 1]
```diff
@@ -89,10 +91,11 @@ export const prepareItemsListForHousehold = async (
         if (listItemError) {
             return { data: null, error: listItemError };
         }
-        itemsList.push({
-            description: row.item_name,
-            ...listItemData,
-        });
+        row.list_type === listType &&
```
Comment (SoftJoelWire) [inline src/pdf/ShoppingList/shoppingListPdfDataProps.ts:94]: this pattern for type checking seems a little strange in a .ts file but seems to work so no real issue. maybe a basic if statement would be better?

File: src/pdf/ShoppingList/shoppingListPdfDataProps.ts [hunk 2]
```diff
@@ -89,10 +91,11 @@ export const prepareItemsListForHousehold = async (
...
+            itemsList.push({
+                description: row.item_name,
+                ...listItemData,
```
Comment (SoftJoelWire) [inline src/pdf/ShoppingList/shoppingListPdfDataProps.ts]: this pattern for type checking seems a little strange in a .ts file but seems to work so no real issue. maybe a basic if statement would be better?

File: src/pdf/ShoppingList/shoppingListPdfDataProps.ts [hunk 3]
```diff
@@ -89,10 +90,11 @@ export const prepareItemsListForHousehold = async (
         if (listItemError) {
             return { data: null, error: listItemError };
         }
-        itemsList.push({
-            description: row.item_name,
-            ...listItemData,
-        });
+        row.list_type === listType &&
```
Comment (Stuart-Knowles-SW) [inline src/pdf/ShoppingList/shoppingListPdfDataProps.ts]: Can we move this before all the processing, so the check is the first thing in the for loop and we don't do any processing if we don't need to? We can use the "Early return pattern" (ask Joel and Leah) and a `continue` statement, so it'll look like this:
```
for (const row of listData) {
        if (row.list_type !== listType) {
            continue
        }
        const { data: listItemData, error: listItemError } = await getQuantityAndNotes(
            row,
            householdSize
        );
        if (listItemError) {
            return { data: null, error: listItemError };
        }
        itemsList.push({
            description: row.item_name,
            ...listItemData,
        });
    }
```

**Review by Stuart-Knowles-SW — COMMENTED** [review]
Looks good, one small change :)

**Review by Stuart-Knowles-SW — APPROVED** [review]
Nice work :)

**Pull request merged**